include/ck/tensor_operation/gpu/device/device_batched_gemm_reduce_xdl_cshuffle.hpp

```diff
@@ -54,6 +54,7 @@ __global__ void
             const ComputeBasePrtOfBatch compute_base_ptr_of_batch_,
             const Block2CTileMap block_2_ctile_map)
 {
+#if (!defined(__HIP_DEVICE_COMPILE__)  || defined(__gfx908__) || defined(__gfx90a__))
```

Comment:
It seems ```!defined(__HIP_DEVICE_COMPILE__)``` could be removed. Could you try it?

Reply:
I added the !defined(__HIP_DEVICE_COMPILE__) to make sure that the code is compiled for host in all cases. Without that flag the device flags are not defined.